backend/monitor/monitors_test.go

```diff
@@ -0,0 +1,185 @@
+
+	monitorName := "testGetMonitorNew"
+	testEntity := types.FixtureEntity("entity")
+	testEvent := types.FixtureEvent(testEntity.ID, "testCheck")
+
+	handler := &testMonitorsHandler{}
+	monitorService := NewService(client, handler, handler)
+	err = monitorService.GetMonitor(context.Background(), monitorName, testEntity, testEvent, 15)
+	require.NoError(t, err)
+
+}
+
+func TestGetMonitorExisting(t *testing.T) {
+	e, cleanup := etcd.NewTestEtcd(t)
+	defer cleanup()
+	client, err := e.NewClient()
+	require.NoError(t, err)
+	defer client.Close()
+
+	monitorName := "testGetMonitorExisting"
+	monitorPath := monitorKeyBuilder.Build(monitorName)
+	testEntity := types.FixtureEntity("entity")
+	testEvent := types.FixtureEvent(testEntity.ID, "testCheck")
+
+	handler := &testMonitorsHandler{}
+	monitorService := NewService(client, handler, handler)
+
+	err = putKeyWithLease(client, monitorPath, 15)
+	require.NoError(t, err)
+
+	err = monitorService.GetMonitor(context.Background(), monitorName, testEntity, testEvent, 15)
+	require.NoError(t, err)
+}
+
+func TestGetMonitorNewTTL(t *testing.T) {
+	e, cleanup := etcd.NewTestEtcd(t)
+	defer cleanup()
+	client, err := e.NewClient()
+	require.NoError(t, err)
+	defer client.Close()
+
+	monitorName := "testGetMonitorNewTTL"
+	monitorPath := monitorKeyBuilder.Build(monitorName)
+	testEntity := types.FixtureEntity("entity")
+	testEvent := types.FixtureEvent(testEntity.ID, "testCheck")
+
+	handler := &testMonitorsHandler{}
+	monitorService := NewService(client, handler, handler)
+
+	err = putKeyWithLease(client, monitorPath, 15)
+	require.NoError(t, err)
+	response, err := client.Get(context.Background(), monitorPath)
+	require.NoError(t, err)
+	fmt.Println("put response in test:", response)
+
+	err = monitorService.GetMonitor(context.Background(), monitorName, testEntity, testEvent, 20)
+	require.NoError(t, err)
+}
+
+func TestLittleGetMonitorNone(t *testing.T) {
+	e, cleanup := etcd.NewTestEtcd(t)
+	defer cleanup()
+	client, err := e.NewClient()
+	require.NoError(t, err)
+	defer client.Close()
+
+	handler := &testMonitorsHandler{}
+	monitorService := NewService(client, handler, handler)
+	mon, err := monitorService.getMonitor(context.Background(), "testLittleGetMonitorNone")
+	require.NoError(t, err)
+	assert.Nil(t, mon)
+}
+
+func TestLittleGetMonitorExisting(t *testing.T) {
+	e, cleanup := etcd.NewTestEtcd(t)
+	defer cleanup()
+	client, err := e.NewClient()
+	require.NoError(t, err)
+	defer client.Close()
+
+	handler := &testMonitorsHandler{}
+	testMon := &monitor{
+		key:     "testLittleGetMonitorExisting",
+		leaseID: 0,
+		ttl:     0,
+	}
+	monitorService := NewService(client, handler, handler)
+	_, err = client.Put(context.Background(), testMon.key, fmt.Sprintf("%d", testMon.ttl))
+	require.NoError(t, err)
+
+	mon, err := monitorService.getMonitor(context.Background(), testMon.key)
+	require.NoError(t, err)
+	assert.EqualValues(t, testMon.key, mon.key)
+}
+
+func TestWatchMonDelete(t *testing.T) {
```

Comment:
Given that these tests don't rely on assertions, they probably would benefit from some comments explaining them.

Reply:
I knew I was forgetting something.